compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -511,20 +503,19 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
                 // If this allocation has size zero, there is no actual mutability here.
                 let (size, _align, _alloc_kind) = self.ecx.get_alloc_info(alloc_id);
                 if size != Size::ZERO {
+                    // Mutable pointer to immutable memory is no good.
                     if ptr_expected_mutbl == Mutability::Mut
                         && alloc_actual_mutbl == Mutability::Not
                     {
                         throw_validation_failure!(self.path, MutableRefToImmutable);
                     }
-                    if ptr_expected_mutbl == Mutability::Mut
-                        && self.ctfe_mode.is_some_and(|c| !c.may_contain_mutable_ref())
-                    {
-                        throw_validation_failure!(self.path, MutableRefInConstOrStatic);
-                    }
-                    if alloc_actual_mutbl == Mutability::Mut
-                        && matches!(self.ctfe_mode, Some(CtfeValidationMode::Const { .. }))
-                    {
-                        throw_validation_failure!(self.path, ConstRefToMutable);
+                    // In a const, everything must be completely immutable.
+                    if matches!(self.ctfe_mode, Some(CtfeValidationMode::Const { .. })) {
```

Comment:
So we don't have any tests testing the promoted code path? Or is that rejected elsewhere even in unleash mode?

Reply:
With promoteds we have to accept this or else we fail [this recently added test](https://github.com/rust-lang/rust/blob/408eeae59d35cbcaab2cfb345d24373954e74fc5/tests/ui/consts/static-promoted-to-mutable-static.rs).

Reply:
🤣 

> We know from the internet

(sorry, I know what you meant, but I literally lolled)

Reply:
:joy: 

(For anyone else reading along, I meant "interner".)